<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>iSEE workshop - package and functionality</title>
    <meta charset="utf-8" />
    <meta name="author" content="Charlotte Soneson, Kevin Rue-Albrecht, Federico Marini, Aaron Lun" />
    <link href="libs/remark-css/default.css" rel="stylesheet" />
    <link href="libs/remark-css/metropolis.css" rel="stylesheet" />
    <link href="libs/remark-css/rladies-fonts.css" rel="stylesheet" />
    <link href="libs/font-awesome/css/all.css" rel="stylesheet" />
    <link href="libs/font-awesome/css/v4-shims.css" rel="stylesheet" />
    <link rel="stylesheet" href="my-theme.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# iSEE workshop - package and functionality
## European Bioconductor Meeting 2020
### <a href="https://csoneson.github.io">Charlotte Soneson</a>, <a href="https://kevinrue.github.io">Kevin Rue-Albrecht</a>, <a href="https://federicomarini.github.io/">Federico Marini</a>, <a href="https://orcid.org/0000-0002-3564-4813/">Aaron Lun</a>
### 2020-12-16 (updated: 2020-12-17)

---


layout: true

&lt;div class="my-header"&gt;&lt;img src="img/stickers/iSEE.png" alt="OxfordiSEE logo" align="right" height="90%"&gt;&lt;/div&gt;

&lt;div class="my-footer"&gt;&lt;span&gt;
iSEE Workshop 2020
&amp;emsp;&amp;emsp;&amp;emsp;&amp;emsp;&amp;emsp;&amp;emsp;&amp;emsp;&amp;emsp;
interactive SummarizedExperiment Explorer
&lt;/span&gt;&lt;/div&gt;







---

# Structure of the session

.pull-left[
## Timeline

&lt;i class="far fa-clock"&gt;&lt;/i&gt;
This session is 90 minutes.

- Introduction slides

- Cloud workshop setup

- Introduction to the GUI

- Interactive exercises
]

.pull-right[
## Interact

&lt;i class="fab fa-slack"&gt;&lt;/i&gt; Slack [#isee](https://community-bioc.slack.com/archives/C8BJLSP8T)

&lt;i class="fab fa-github-square"&gt;&lt;/i&gt; Slides [GitHub Pages](https://isee.github.io/iSEEWorkshopEuroBioc2020Slides/#1)

&lt;i class="fab fa-github-square"&gt;&lt;/i&gt; Workshop vignette [GitHub Pages](https://isee.github.io/iSEEWorkshopEuroBioc2020)


&lt;i class="fab fa-github"&gt;&lt;/i&gt;
GitHub [organization](https://github.com/iSEE)
]

---

# &lt;i class="fas fa-users"&gt;&lt;/i&gt; The team

&lt;br/&gt;

&lt;table&gt;
&lt;tr&gt;
&lt;td width="25%"&gt;&lt;a href="https://kevinrue.github.io"&gt;&lt;img src="img/people/kevin-rue.jpg"&gt;&lt;/a&gt;&lt;/td&gt;
&lt;td width="25%"&gt;&lt;a href="https://csoneson.github.io"&gt;&lt;img src="img/people/charlotte-soneson.png"&gt;&lt;/a&gt;&lt;/td&gt;
&lt;td width="25%"&gt;&lt;a href="https://federicomarini.github.io"&gt;&lt;img src="img/people/federico-marini.png"&gt;&lt;/a&gt;&lt;/td&gt;
&lt;td width="25%"&gt;&lt;a href="https://orcid.org/0000-0002-3564-4813"&gt;&lt;img src="img/people/aaron-lun.png"&gt;&lt;/a&gt;&lt;/td&gt;
&lt;/tr&gt;
&lt;tr&gt;
&lt;td align='center' width="25%"&gt;Kevin&lt;/td&gt;
&lt;td align='center' width="25%"&gt;Charlotte&lt;/td&gt;
&lt;td align='center' width="25%"&gt;Federico&lt;/td&gt;
&lt;td align='center' width="25%"&gt;Aaron&lt;/td&gt;
&lt;/tr&gt;
&lt;/table&gt;



???

This was a team effort!

---

# &lt;i class="fas fa-pencil-ruler"&gt;&lt;/i&gt; Designed for Bioconductor

&lt;img src="img/stickers/biocstickers.jpg" width="75%" style="display: block; margin: auto;" /&gt;

&lt;https://bioconductor.org&gt;

???

iSEE tightly integrates with other packages of the Bioconductor project.

---

# SingleCellExperiment

&lt;img src="img/OSCA/bioc-figures_v2-02.png" width="1412" height="400px" style="display: block; margin: auto;" /&gt;

.footnote[

]

???

iSEE focuses on the SingleCellExperiment class.
This class stores all the data and metadata associated with assays, cells, and features.

---

# SingleCellExperiment usage in a workflow

.pull-left[
- Data import (`DropletUtils`, `tximeta`)

- Quality control (`scater`)

- Normalization, feature selection (`scran`, `zinbwave`)

- Dimensionality reduction (`BiocSingular`, `scater`, `zinbwave`)

- Clustering (`SC3`, `clustree`)

- Marker gene detection (`scran`, `scDD`)

- Trajectory inference (`slingshot`)

- Visualization (`iSEE`)

- And much more...
]

.pull-right[
&lt;img src="img/OSCA/bioc-figures_v2-03.png" width="1960" style="display: block; margin: auto;" /&gt;

.right[

]
]

???

The SingleCellExperiment class is designed to accomodate all the information produced along a typical single-cell analysis workflow.

Those data include raw data:

- raw assay data
- experimental metadata
Processed data:
- quality control metrics
- normalized data
- dimensionality reduction results
Downstream analyses:
- cluster labels
- differential expression results
- downstream cell and feature annotations

---

# iSEE &lt;i class="fas fa-heartbeat"&gt;&lt;/i&gt; SingleCellExperiment

&lt;img src="img/OSCA/OSCA-figure-4.png" width="60%" style="display: block; margin: auto;" /&gt;

.footnote[

]

???

Here we demonstrate how iSEE dissects SCE objects to produce figures.

---

# The iSEE interface

&lt;img src="img/screenshots/iSEE_screenshot.png" width="600px" height="500px" style="display: block; margin: auto;" /&gt;

???

iSEE provides a powerful yet flexible user interface that includes 8 predefined panel types.
That said, it also gives the freedom to define any number custom panel types, both plots and tables.

---

# &lt;i class="fas fa-laptop"&gt;&lt;/i&gt; Workshop

&lt;img src="img/misc/workshop.png" width="400px" height="200px" style="display: block; margin: auto;" /&gt;

## To get started

- Log in to [http://app.orchestra.cancerdatasci.org/](http://app.orchestra.cancerdatasci.org/)

- Choose the `iSEE: Interactive SummarizedExperiment Explorer` workshop and launch the RStudio session

- In a separate tab, open the [pkgdown website](https://isee.github.io/iSEEWorkshopEuroBioc2020/) - vignettes are available under 'Articles'

???

With that introduction to the user interface, let us head into the workshop!

Generated by: &lt;i class="fab fa-github"&gt;&lt;/i&gt; [kevinrue/magick-profile](https://github.com/kevinrue/magick-profile)

- [Interface overview](https://isee.github.io/iSEEWorkshopEuroBioc2020/articles/overview.html),
  or `vignette('overview', 'iSEEWorkshopEuroBioc2020')`

- [Exercises/"recipes"](https://isee.github.io/iSEEWorkshopEuroBioc2020/articles/recipes.html),
  or, `vignette('recipes', 'iSEEWorkshopEuroBioc2020')`
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
